sapp/ui/frontend/src/Filter.js

```diff
@@ -780,7 +784,65 @@ const SaveFilterModal = (
     return <Alert type="error">{error.toString()}</Alert>;
   }
 
-  const onOk = (): void => {
+  let filterList = [];
+
+  const importProps = {
+    multiple: true,
+    onRemove: filterFile => {
+      filterList.splice(filterList.indexOf(filterFile), 1);
+    },
+    beforeUpload: filterFile => {
+      if (filterFile.type !== 'application/json') {
+        message.error(`${filterFile.name} is not a json file`);
+      } else {
+        filterList = [...filterList, filterFile];
+      }
+      return filterFile.type === 'application/json' ?
+        false : Upload.LIST_IGNORE;
+    },
+    filterList,
+  };
+
+  const onImport = () => {
+    let count = filterList.length;
+    let skipped = 0;
+    filterList.forEach(filterFile => {
+      let reader = new FileReader();
+      reader.onload = (e) => {
+        let filter: FilterDescription = JSON.parse(e.target.result);
+        if (filter["name"] === undefined ||
```

Comment:
Instead of filtering for which files in `filterList` we can import in the same loop as importing filters, could we use `Array.prototype.filter()` to filter invalid files first, then run the for each loop below on the valid list of filters?

The callback function we use in `Array.prototype.filter()` to validate if a filter file is valid can be reused to perform validation before we export filters in https://github.com/facebook/sapp/pull/31

Reply:
Hey nice idea but the problem is the fileReader which is kinda not straight flowing with its onload function. So I did try using it on the beforeUpload function, but even with Promises I couldn't get it to work properly, hence I put the validation here.

Reply:
> Nice idea but the problem is the fileReader which is kinda not straight flowing with its onload function. 

@abishekvashok Sorry I'm not following - what was the issue with `FileReader`'s `onload` property? Was the issue related to the `load` event not firing when you included it in the `callback` function for `Array.prototype.filter()`? 

Would something roughly like this not work?
```javascript
const filterInvalidFilterFiles(filterFile) {
	let reader = new FileReader();
	reader.onload = (e) => {
        let filter: FilterDescription = JSON.parse(e.target.result);
        if (filter["name"] !== undefined &&
        	      filter["description"] !== undefined &&
          Object.keys(filter).length > 2) {
        	return true;
	}
    	message.error(`Invalid filter file, skipping import of ${filterFile.name}`);
    	return false;
	}
}

const onImport = () => {
       ...
       filterList.filter(filterInvalidFilterFiles)

```

Reply:
Yep, this will return true everytime, I pushed this in beforeUpload but I was always getting true.

Reply:
Hmm return inside onload, makes sense, I will try this :) Thank you

Reply:
I tried it inside of beforeUpload, and appending the result to filterList to so that the reader can be used only once, but it doesn't work - the same result :/

Reply:
I will try a separate filter function now, as provided, but we will need to read the file two times!

Reply:
@0xedward Did try, not working always returning true :/

Reply:
Uff, I was being dump :| the problem is that reader.onload is a function. So when we return true inside of the function -> it is the value returned from reader.onload() and not our `removeInvalidFilterFiles(filterFile)` function. So whatever reader.onload() does return, our `removeInvalidFilterFiles(filterFile)` function always returns void.